src/config.c

```diff
@@ -1599,9 +1584,15 @@ int rewriteConfig(char *path, int force_write) {
      * the rewrite state. */
 
     /* Iterate the configs that are standard */
-    for (standardConfig *config = configs; config->name != NULL; config++) {
+    dictIterator *di = dictGetIterator(configs);
+    dictEntry *de;
+    while ((de = dictNext(di)) != NULL) {
+        standardConfig *config = dictGetVal(de);
+        /* Only rewrite the primary names */
+        if (!(config->flags & PREFERRED_CONFIG)) continue;
```

Comment:
It feels like wrote the opposite.
```suggestion
        if (config->flags & PREFERRED_CONFIG) continue;
```

Reply:
Ignore,Already fixed by the last commit.